src/PackagePublisher/Program.cs

```diff
@@ -24,7 +24,9 @@ public static class Program
 
         // Do NOT log the key
         private static string _apiKey = null;
-        private static int _maxRetryCount = 10;
+        private static int _maxRetryCount = 5;
+        private static int _packagePushTimeoutInSeconds = 90;
+        private static int _semaphoreTimeoutInSeconds = _maxRetryCount * _packagePushTimeoutInSeconds;
```

Comment:
First, I would change both of these from `int *InSeconds` to `TimeSpan *`.  I agree the value should be at least `_maxRetryCount * _packagePushTimeout * NumPackages`.  Or, just choose a huge value that should always be sufficient for all packages to finish, say 1 hour.